aea/configurations/base.py

```diff
@@ -1845,18 +1850,18 @@ def json(self) -> Dict:
                 "skills": sorted(map(str, self.skills)),
                 "default_connection": self.default_connection,
                 "default_ledger": self.default_ledger,
-                "logging_config": self.logging_config,
+                "default_routing": {
+                    str(key): str(value) for key, value in self.default_routing.items()
+                },
+                "connection_private_key_paths": self.connection_private_key_paths_dict,
                 "private_key_paths": self.private_key_paths_dict,
+                "logging_config": self.logging_config,
                 "registry_path": self.registry_path,
                 "component_configurations": self.component_configurations_json(),
             }
         )  # type: Dict[str, Any]
 
-        if len(self.connection_private_key_paths_dict) > 0:
```

Comment:
both default_routing and connection_private_key_paths are important configurations of the agent behaviour. So they need to be present.